e2e_tests/tests/requirements.txt

```
@@ -5,7 +5,8 @@ pytest-timeout
 pexpect
 torch==1.9.0
 torchvision==0.10.0
-tensorflow==2.4.3
+tensorflow==2.4.3; sys_platform != 'darwin'
+tensorflow-macos==2.7.0; sys_platform == 'darwin'
```

This is cool! I didn't know `requirements.txt` could do that.

I was trying it out and it seems `tensorflow-macos` for x86_64 has only a build for py38. Should we perhaps do this for arm64 only, i.e.:
```suggestion
tensorflow==2.4.3; sys_platform != 'darwin' or platform_machine != 'arm64'
tensorflow-macos==2.7.0; sys_platform == 'darwin' and platform_machine == 'arm64'
```